src/falcor/transforms/index.js

```diff
@@ -196,3 +196,21 @@ export function remove ( collection, user, _id ) {
     });
 }
 
+
+export function archiveDocument(collection, _id, userId) {
+  return this.flatMap(db => {
+    db = db.mongo.collection(collection);
+    return Observable.fromPromise(db.findOneAndUpdate(
```

Comment:
On a `flatMap`, I don't think you need to call `Observable.fromPromise`. I think it should detect that it is a promise and resolve it automatically as part of the `flatMap`.